File: client/src/lib/types/context.ts
import type {
	AnagramsPlayerData,
	ClientInfo,
	Games,
	PostGameInfo,
	RoomSettings,
	Uuid,
	WordBombPlayerData
} from './messages';

export type State = 'Connecting' | 'Error' | 'Lobby' | Games;

export type ConnectionData = {
	room: string;
	settings: RoomSettings;
	uuid: Uuid;
	roomOwner: Uuid;
	clients: Array<ClientInfo>;
	chatMessages: Array<[string, string] | string>;
};

export type LobbyData = {
	ready: Array<Uuid>;
	startingCountdown: number | null;
	postGame: PostGameInfo | null;
};

export type WordBombData = {
	players: Array<WordBombPlayerData>;
	turn: Uuid;
	prompt: string;
	input: string;
	usedLetters: Set<string> | null;
};

export type AnagramsData = {
	players: Array<AnagramsPlayerData>;
	anagram: string;
};

File: crates/server/src/room/context.rs
use crate::{
    messages::RoomSettings,
    room::{clients::Clients, sender::RoomSender},
};

pub struct Context<'a> {
    pub room: &'a RoomSender,
    pub clients: &'a mut Clients,
    pub settings: &'a mut RoomSettings,
}

impl<'a> Context<'a> {
    pub fn new(
        room: &'a RoomSender,
        clients: &'a mut Clients,
        settings: &'a mut RoomSettings,
    ) -> Self {
        Self {
            room,
            clients,
            settings,
        }
    }
}
